### ISSUE TITLE
implement a more idiomatic (HTML) templating solution

### ISSUE THREAD

**Issue by thescientist13** [ProjectEvergreen/greenwood#1133]
## Type of Change
Enhancement

## Summary
Coming out of #1126 , it's clear that how Greenwood handles various aspects of its templating, primarily for HTML, is a little too naive / brittle.

For example, Greenwood uses placeholders for injecting content into an HTML template
```html
<html>
  <body>
    <content-outlet><content-outlet>
  </body>
</html>
```

Would get handled like this in the CLI
```sh
body = body.replace(/\<content-outlet>(.*)<\/content-outlet>/s, processedMarkdown.contents);
```

This issue is that [`$1` is used for matching](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#specifying_a_string_as_the_replacement) when used in conjunction with the String `.replace` function.

So we have to keep doing things like
```js
body = body.replace(/\<content-outlet>(.*)<\/content-outlet>/s, processedMarkdown.contents.replace(/\$/g, '$$$'));
```

## Details
So rather than keep suffering through these issues...
- #1126 
- #743
- #657

Since we are already using an [HTML parser](https://www.npmjs.com/package/node-html-parser), I see it supports methods like [`set_content`](https://github.com/taoqf/node-html-parser#set_contentcontent-string--node--node) which could a much more practical way to approach this problem.

So basically any [usage of `.replace`](https://github.com/search?q=repo%3AProjectEvergreen%2Fgreenwood%20.replace&type=code), for HTML at minimum, would be a candidate for refactoring.

**Comment by thescientist13** [comment]
Seeing similar issues as per #1241
